exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -118,7 +118,7 @@ func mapMetricToSplunkEvent(res pcommon.Resource, m pmetric.Metric, config *Conf
 			{
 				fields := cloneMap(commonFields)
 				populateAttributes(fields, dataPt.Attributes())
-				fields[metricFieldName+sumSuffix] = dataPt.Sum()
+				fields[metricFieldName+sumSuffix] = sanitizeFloat(dataPt.Sum())
```

Comment:
In general should not be NaN (may be if dataPt.Flags().NoRecordedValue() == true), check also for "HasSum()" (it is 0 if not present).

Reply:
It is throwing the following errors in the customer environment. 
Error: 
```
Permanent error: dropped metric event: &{0xc030652cd0 aks-agentpool-34656813-vmss_9 kubernetes  aks_metrics_fni metric map[azure.resourcegroup.name:XYZ azure.vm.name:aks-agentpool-34656813-vmss_9 azure.vm.scaleset.name:aks-agentpool-34656813-vmss azure.vm.size:Standard_D8as_v5 cloud.account.id:ID cloud.platform:azure_aks cloud.provider:azure cloud.region:westeurope deployment.environment:qas host.id:8bb51409-5e87-44db-95d5-4781e00700bd http.scheme:http k8s.cluster.name:cluster k8s.namespace.name:istio-system k8s.node.name:aks-agentpool-34656813-vmss000009 k8s.pod.name:istio-ingressgateway-internal-7dd4b7f89d-5vgfj k8s.pod.uid:fcfeecd1-6b94-44c4-af8e-8b13a59deb9b metric_name:istio_agent_go_sched_latencies_seconds_sum:NaN metric_type:Histogram net.host.name:HOST net.host.port:15020 os.type:linux service.instance.id:HOST:15020 service.name:prometheus_simple/HOST:15020]}, error: splunk.Event.Fields: unsupported value: NaN
```

Reply:
Customer here. Let me know if you need any additional analysis/logs/whatever.

Reply:
The HEC field shouldn't be reported if the sum isn't set. That can be checked with `dataPt.HasSum()`, adding that condition should solve the problem.

Reply:
Sorry for the silly question, I haven't worked with metric data. Should I only skip the sum metric(count, or every metric related to the data point if `dataPt.HasSum()` returns false?

Reply:
> Sorry for the silly question, I haven't worked with metric data. Should I only skip the sum metric or every metric related to the data point if dataPt.HasSum() returns false?

Yes, please

Reply:
Thanks for your input. `dataPt.HasSum()` returns true even if sum in `NaN`. So, I have added validation for `NaN` for the sum.

Reply:
Thanks for your input. `dataPt.HasSum()` returns true even if sum in `NaN`. So, I have added validation for `NaN` for the sum.

Reply:
Which input/receiver do you use that produce `NaN` sums?

Also what about `dataPt.Flags().NoRecordedValue()` ? Is that flag set to true?

Reply:
> Which input/receiver do you use that produces NaN sums?

From the logs, it looks like `prometheus_simple` is producing that metric.
> Also what about dataPt.Flags().NoRecordedValue() ? Is that flag set to true?

I haven't checked it yet.

Reply:
@hailpis1 can help us in reproducing the issue. 
From my understanding, `NaN` sum is coming from the scraping endpoint. And Prometheus receiver is collecting it as it is. What is expected from the Prometheus receiver when it receives such metrics?

Meanwhile, if the current changes look good, can we merge them? @hailpis1 is waiting on this.

Reply:
Sure it looks good to me